owrx/websocket.py

```diff
@@ -175,60 +177,67 @@ def read_loop(self):
         def protected_read(num):
             data = self.handler.rfile.read(num)
```

Comment:
all reading should be moved into the loop; just initialize `data` to an empty array and start with a select. that way the code should be easier to read.